sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java

```diff
@@ -259,6 +298,64 @@ public int hashCode() {
     }
   }
 
+  interface ColumnPosition {
+    First FIRST = new First();
```

Comment:
It's awkward for this to expose `ColumnPosition.FIRST` and `ColumnPosition.after`. Can you add a static method `first` that returns `First.INSTANCE` instead?

Also, if this is to be a singleton then the constructor for `First` should be private, as should `First.INSTANCE`.